core/graph/utils/extract_keywords.py

```
@@ -0,0 +1,282 @@
+            with torch.no_grad():
+                outputs = self.model(**inputs)
+                # Mean pooling
+                batch_embeddings = outputs.last_hidden_state.mean(dim=1)
+                # Normalize
+                batch_embeddings = F.normalize(batch_embeddings, p=2, dim=1)
+                
+            embeddings.append(batch_embeddings.cpu().numpy())
+        
+        return np.vstack(embeddings) if embeddings else np.array([])
+    
+    def process_papers(self, limit: int = None):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)